src/test/java/bgu/spl/mics/MessageBrokerTest.java

```diff
@@ -1,19 +1,105 @@
 package bgu.spl.mics;
 
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
 public class MessageBrokerTest {
+    TestSubscriber s1;
+    TestSubscriber s2;
+    MessageBroker m;
+
     @BeforeEach
-    public void setUp(){
+    public void setUp() {
+        m = MessageBrokerImpl.getInstance();
+        s1 = new TestSubscriber("s1");
+        s2 = new TestSubscriber("s2");
+    }
+
+    @AfterEach
+    public void tearDown() {
+        m.unregister(s1);
+        m.unregister(s2);
+    }
 
+    @Test
+    public void testSingleton() {
+        MessageBroker otherInstance = MessageBrokerImpl.getInstance();
+        assertTrue(otherInstance == m, "different instances of singleton MessageBroker exist!");
+    }
+
+    @Test
+    public void testEventSubscribe() {
+        m.sendEvent(new IntEvent(0));
+        m.sendEvent(new IntEvent(1));
+        safeAwaitMessage(true);
```

Comment:
move deceleration above this method call

Reply:
Do you mean the declaration of safeAwaitMessage?

Reply:
Yes. We always prefer a method definition appear before it is used

Reply:
Done